### ISSUE TITLE
ComboBox should use a line instead of a separator

### ISSUE THREAD

**Issue by marlitas** [phetsims/sun#810]
From Dev Meeting 11/17/22:

> @jonathanolson:  I'm not convinced ComboBoxButton should be using a separator. That seems fragile and should be done with something else.

This will remove the deprecated separator and use Line or a similar component to achieve the proper look and layout for comboBox.

**Comment by marlitas** [comment]
This work is completed in the above commits. Assigning to @jbphet for review  as responsible dev.

**Comment by jbphet** [comment]
The changes look reasonable to me.  I also tested the behavior in the sun demo and in Molarity, which has a large and prominent combo box, and the behavior and appearance seem correct.  Closing.

**Comment by pixelzoom** [comment]
Reopening.  I'm seeing buggy layout when the combo box button grows.  For example, here's the control panel in the Discrete screen of Fourier, after pressing RIGHT arrow once with `stringTest=dynamic`. 

<img width="267" alt="screenshot_2044" src="https://user-images.githubusercontent.com/3046552/206918780-334d61b7-3d41-4c67-b5e7-98481eb42b12.png">

Blocks publication of any sim that uses ComboBox and supports dynamic layout.

**Comment by pixelzoom** [comment]
I do not see this problem with the combo boxes in pH Scale or Beer's Law Lab.

**Comment by marlitas** [comment]
It looks like `AlignBox` is not receiving the correct `minimumWidth` from it's `GridBox` content in Fourier. 

AlignBox minimumWidth = 135
![Screenshot 2022-12-12 at 11 07 18 AM](https://user-images.githubusercontent.com/42476338/207133918-4059409b-6c7b-46ae-ab85-69741f003728.png)

GridBox minimumWidth = 148
![Screenshot 2022-12-12 at 11 07 33 AM](https://user-images.githubusercontent.com/42476338/207133842-5e5da893-7fff-4e1e-8f33-03e4ac49502b.png)

I confirmed this was happening prior to switching out VSeparatorDeprecated with Line in ComboBoxButton. Since this seems unrelated to this conversion I created a new issue (https://github.com/phetsims/sun/issues/816), and will close this one as completed.
